### ISSUE TITLE
Show recent finds / DNF on stored list view

### ISSUE THREAD

**Issue by DaveThistlethwaite** [cgeo/cgeo#12759]
### Tell us your idea!

On the list view of stored caches it would be useful to show the last, say, 4 finds as yellow smileys or blue frowns as GSAK does

Maybe drop the word 'premium' from the view to free up real estate

### What solution would you suggest?

Below the code and D/T rating row would be a new row showing the recent find emojis

**Comment by MagpieFourtyTwo** [comment]
Like the idea.  Although for most Stored lists, the information will most probably not be up to date and for Live/Online lists the needed data (resp. the logs) is not available when building the list (AFAIK).

BTW: Dropping the phrase "Premium" would of course be a way to free some space (would at least make sense for Premium members, as for those it's informativeness appears to be pretty limited), but it would not be a real solution, as the phrase does a) not always appear, and b) is below the name of the cache, i. e. at a already pretty crowded place (with e. g. the list name(s).

Therefore I would place the log type indicators on the very right, below distance and direction to the cache. Over there esp. offline stored caches have some free room, because of the list of lists below the name.

**Comment by SammysHP** [comment]
A simple indicator for found / not found would be enough. Smaller than smileys, maybe just little dots.

**Comment by wolverine007** [comment]
Like the idea as well, stating as much as possible even inside a list is always a great idea, but in this case it is necessary that the cache has been refreshed before, elsewise you might see data which is several weeks old.

So we are coming back to a request I placed several month ago - automatically refresh lists basing on a scheduled refresh task, e. g. during each night or similar.


But even without a refreshed list we might have issues to displays the smileys. Just picked a random list of stored caches and as you can see, each one is stating different stuff resulting in different hights, so I placed the smileys to random places - for the last one was not even enough space available:

![list_smilies.jpg](https://user-images.githubusercontent.com/4508228/154630105-2282c439-7db6-4298-955f-86e574c83db4.jpg)

Placing them behind the cache title above distance and direction seems to be a valid place for me, but needs a fixed hight than.

**Comment by MagpieFourtyTwo** [comment]
Yep, see it the same:  Not-up-to-date-data will be *the* major problem when showing log type indicators.
Which is in fact the reason why I would suggest to either ...
- *not* show them at all (even with a weeping eye),
- or to just show them if the corr. cache has been refreshed within let's say the last 12 hours ...?
- or if the cache data is older, show the smileys/frownies greyed out or with a reduced opacity ...?

As for automatic updates: Lists can grow to infinity, so updating complete lists should be an absolute no-go, IMNSHO.
Not just because of limited data plans and bad coverage (update could be restricted to be performed when being in WLAN only),
but esp. WRT server load, which would massively increase (and would give c:geo an even harder stand at some services),
even more as AFAIK (please correct me if I'm wrong) the current list items are based on a "light" server request (i. e. need less data), while with logs a complete request would be necessary.

And even (automatically) updating just the visible portion of a list could be a performance issue, esp. if data coverage is bad.
Moreover, imagine what will happen behind the scenes, if a user decides to scroll though a list with 20k caches ... \*brrr\*

WRT the representation (if it will be implemented): I would highly prefer the "outer three examples", with the indicators above distance or below cardinal direction. While the latter would fit even a bit better, IMO, as the indicators are in fact a refinement of the log counter, thus it would be more logical to have them together as close as possible (e. g. just one smiley on the very right top could easily be misunderstood as "Found by me" indicator, esp. by inexperienced users).

Moreover, there should not just be smileys and frownies but esp. symbols for Owner Maintenance and Notes, too, as these are often even more important than a Found). Therefore I would - if at all - implement all Log types.

**Comment by DaveThistlethwaite** [comment]
Thank you for the positive comments on my first feature request. Not knowing the architecture I was ignorant of the issue about the list not being updated but personally I don't see that as a problem - if I click on a cache I don't know if the same representation there is up to date or not.  I usually store an offline list and refresh it before a day in the field

Of the presented mockups above I prefer the first one. I agree with the comment about all log types too.

**Comment by ztNFny** [comment]
> Like the idea as well, stating as much as possible even inside a list is always a great idea, but in this case it is necessary that the cache has been refreshed before, elsewise you might see data which is several weeks old.
> 
> So we are coming back to a request I placed several month ago - automatically refresh lists basing on a scheduled refresh task, e. g. during each night or similar.
> 
> But even without a refreshed list we might have issues to displays the smileys. Just picked a random list of stored caches and as you can see, each one is stating different stuff resulting in different hights, so I placed the smileys to random places - for the last one was not even enough space available:
> 
> ![list_smilies.jpg](https://user-images.githubusercontent.com/4508228/154630105-2282c439-7db6-4298-955f-86e574c83db4.jpg)
> 
> Placing them behind the cache title above distance and direction seems to be a valid place for me, but needs a fixed hight than.

A more space-saving way to implement such a feature would be a single icon that shows the logs as segments, e.g. a circle filling, similar to the DT markers.  That one coule (optionally) replace the existing FP count without any chnage in layout even.

**Comment by ztNFny** [comment]
Proposal for displaying the last 5 logs in a more compact way:

![image](https://user-images.githubusercontent.com/1258173/159502151-1f4a6a36-3cba-42e6-b095-29fdb0ef1018.png)

**Comment by MagpieFourtyTwo** [comment]
Like the idea.  And the proposal, too.  Esp. for "Live" caches.
WRT offline stored caches, I would "limit" the information to, let's say, reasonably recent ones ...
Perhaps show it just for caches which have been stored/refreshed up to a week ago ...?
Would additionally be a nice reminder to refresh the offline stored listing ...  ;)

**Comment by ztNFny** [comment]
> Like the idea. And the proposal, too. Esp. for "Live" caches.

For live map this won't be available as it'd require loading additional data for each cache, thereby slowing down live map significantly

**Comment by MagpieFourtyTwo** [comment]
Ah, the colors: Esp. green and red should be as different as possible:
[Red-Green Color Blindness](https://www.color-blindness.com/red-green-color-blindness): "About 8% of all men and 0.5% of all women are suffering from it."

**Comment by MagpieFourtyTwo** [comment]
> live map 

Sorry for being imprecise: Thought of live lists.

**Comment by MagpieFourtyTwo** [comment]
Nevertheless you're of course right, sacrificing performance would not be any good.

**Comment by ztNFny** [comment]
> Sorry for being imprecise: Thought of live lists.

Live maps, nearby, search are the same, only basic cache info is available - logs are not loaded until the full listing is loaded

**Comment by Lineflyer** [comment]
Just my few cents from out of Spain:

- The feature would be a nice adddon
- I dont like the colored only approach
- Cant we shrink the smiley for the last 3-5 logs to fit below the fav points (like in the color version)? Due to the distinct coloring e.g. for DNF vs. Found it would still be recognizable IMHO

**Comment by murggel** [comment]
@DaveThistlethwaite @MagpieFourtyTwo  I think we can close here? It was introduced some time ago for offline lists
see #14084

**Comment by moving-bits** [comment]
Closing here, as feature got implemented with #14152 in 2023
